src/components/tabs/__internal__/tab-title/tab-title.component.js

```diff
@@ -104,8 +125,12 @@ const TabTitle = React.forwardRef(
         isInSidebar={isInSidebar}
         {...tabTitleProps}
         {...tagComponent("tab-header", tabTitleProps)}
+        onKeyDown={handleKeyDown}
       >
         <StyledTitleContent
+          as={href && "a"}
```

Comment:
Initialize ``const hasHref = !!href`` at the top to make it a litttle more verbose and avoid possible xss.